native_toplevel/opttoploop.mli

```diff
@@ -161,3 +161,27 @@ val override_sys_argv : string array -> unit
    This is called by [run_script] so that [Sys.argv] represents
    "script.ml args..." instead of the full command line:
    "ocamlrun unix.cma ... script.ml args...". *)
+
+(* JIT hook *)
+
+type res = Ok of Obj.t | Err of string
```

Comment:
This type appears to be unused

Reply:
It is used by the JIT. See: https://github.com/NathanReb/ocaml-jit/blob/main/lib/externals.mli#L25 and https://github.com/NathanReb/ocaml-jit/blob/main/lib/jit.ml#L137 (which is the JIT equivalent of [`Opttoploop.dll_run`](https://github.com/ocaml-flambda/flambda-backend/blob/main/native_toplevel/opttoploop.ml#L49)).

Reply:
Sure, but if it's not used in this interface, it should be in the JIT library, no?

Reply:
It could be moved there. It felt safer to me to use whatever the upstream toplevel was using so that if it ever changes we'd know right away but that seems unlikely.

I'm happy to move it to the JIT janestreet branch if you prefer.

Reply:
Ah I see, this type is introduced by the upstream toplevel unification?

Reply:
No it was there before an is only used internally by the native toplevel in the unified version. Looking at it, it is true that its use is pretty self-contained so we could redefine that type in the JIT and it shouldn't cause any trouble.

My main concern was to closely stick to the native toplevel plumbing so I reused as much of its code as possible. I'm a little worried that if we use a copy of the code as it is now rather than using toploop functions and types, it might not age well and evolve at different speeds.

A better long-term solution might be to expose better hooks in the toplevel interface so that we don't have to re-implement functions such as `jit_run`. In the meantime I'm ok to un-expose this and add it to the JIT as an internal.